traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/config/ConfigHandler.java

```diff
@@ -491,21 +491,36 @@ private void parseTopologyConfig(final JsonNode allTopologies, final Map<String,
 
 		deliveryServiceMap.forEach((xmlId, ds) -> {
 			final List<DeliveryServiceReference> dsReferences = new ArrayList<>();
+			final List<String> dsNames = new ArrayList<>(); // for stats
 			Stream.of(ds.getTopology())
-					.filter((topologyName) -> !Objects.isNull(topologyName) && topologyMap.containsKey(topologyName))
-					.flatMap((topologyName) -> topologyMap.get(topologyName).stream())
-					.flatMap((node) -> cacheRegister.getCacheLocation(node).getCaches().stream())
-					.filter((cache) -> ds.hasRequiredCapabilities(cache.getCapabilities()))
-					.forEach((cache) -> {
+					.filter(topologyName -> !Objects.isNull(topologyName) && topologyMap.containsKey(topologyName))
+					.flatMap(topologyName -> {
+						statMap.put(ds.getId(), dsNames);
+						return topologyMap.get(topologyName).stream();
+					})
+					.flatMap(node -> cacheRegister.getCacheLocation(node).getCaches().stream())
+					.filter(cache -> ds.hasRequiredCapabilities(cache.getCapabilities()))
+					.forEach(cache -> {
+					    cacheRegister.getDeliveryServiceMatchers(ds).stream()
+								.flatMap(deliveryServiceMatcher -> deliveryServiceMatcher.getRequestMatchers().stream())
+								.map(requestMatcher -> requestMatcher.getPattern().pattern())
+								.forEach(pattern -> {
+									final String remap = ds.getRemap(pattern);
+									final String fqdn = pattern.contains(".*") && !ds.isDns()
+											? cache.getId() + "." + remap
+											: remap;
+									dsNames.add(getDsName(fqdn, tld));
+									try {
+										dsReferences.add(new DeliveryServiceReference(ds.getId(), fqdn));
```

Comment:
This seems to be a little bit different from the legacy (deliveryservice-server) behavior in `parseCacheConfig`. That legacy behavior seemed to only add the "standard" `routingName.xmlId.cdnDomain` FQDN for DNS deliveryservices or the `hostName.xmlId.cdnDomain` FQDN for HTTP deliveryservices into `dsReferences` -- not the "alias" FQDNs. That is because the first server-deliveryService reference is never the "alias" FQDN. As far as I can tell, it is only the `statTracker` that seems to need the "alias" FQDNs.

Basically, if a cache "cache01" belongs to a topology that has the given delivery services assigned:
```
xml_id = foo, routing_name = foo, routing_type = HTTP, 0th pattern = .*\.foo\..*
xml_id = bar, routing_name = bar, routing_type = DNS, 0th pattern = .*\.bar\..*
```
I would expect only the following FQDNs to be added to the cache's `DeliveryServiceReferences`:
```
cache01.foo.cdn.example.com
bar.bar.cdn.example.com
```
The aliases, if the delivery services have any, would *not* be added to the cache's `DeliveryServiceReferences`, but would only be added to the `dsNames` in the `statMap` (in addition to the aforementioned "standard" FQDNs).

Reply:
Not creating `DeliveryServiceReference`s for aliases in 2e509d420 (but still adding them to `statMap`).

Reply:
Ok, this is better, but it still doesn't seem to account for DNS-routed delivery services. Referring to my example above, it _appears_ to be adding `xmlId.cdnDomain` to `dsReferences`, when really I think that is supposed to be `routingName.xmlId.cdnDomain`. From my example, it should add `bar.bar.cdn.example.com` not `bar.example.cdn.com`.

Does that make sense? If I create that DNS-routed delivery service, assign it to a Topology, snapshot, then `dig @localhost -p1053 bar.bar.example.cdn.com a`, I should get a NOERROR response back with the cache IP addresses, but I'm getting NXDOMAIN. Are you getting proper responses back?

Reply:
Oops, it was prepending the xmlId for DNS DSes. Changed it in 35f96f518 to prepend the routing name instead.

> Are you getting proper responses back?

Not for a DNS DS, I missed that detail in the examples you provided.

Reply:
The alias-detecting condition was catching DNS DS URLs, fixed in 18f2f3a15. I am getting proper responses back for DNS DSes.